### PR TITLE
Remove the client transport profile filter

### PR DESCRIPTION
**Pull request by jaymode**
Now that the transport client has been removed, the client transport
profile filter can be removed from security. This filter prevented node
actions from being executed using a transport client.

### CODE REVIEW

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ServerTransportFilterTests.java [hunk 1]
```diff
@@ -200,21 +186,10 @@ public void testNodeProfileAllowsNodeActions() throws Exception {
...
-
-    private ServerTransportFilter.ClientProfile getClientFilter(boolean reservedRealmEnabled) throws IOException {
-        Settings settings = Settings.builder().put("path.home", createTempDir()).build();
-        ThreadContext threadContext = new ThreadContext(settings);
-        return new ServerTransportFilter.ClientProfile(authcService, authzService, threadContext, false, destructiveOperations,
-                reservedRealmEnabled, new SecurityContext(settings, threadContext), new XPackLicenseState(settings));
-    }
-
-    private ServerTransportFilter.NodeProfile getNodeFilter(boolean reservedRealmEnabled) throws IOException {
+    private ServerTransportFilter getNodeFilter() {
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/ServerTransportFilterTests.java]: Nit: Rename method?
```suggestion
    private ServerTransportFilter getTransportFilter() {
```

**Review by tvernum — APPROVED** [review]
LGTM

**Review by jkakavas — APPROVED** [review]
LGTM

**Comment by tvernum** [comment]
@elasticmachine update branch

**Pull request merged**